base/common/system.py

```diff
@@ -48,10 +49,19 @@ def _remove_heading_from_df_output(df_output: bytes) -> int:
         LOG.info(f"obtaining free space on bu hdd with command: {' '.join(command)}. Received {free_space_on_bu_hdd}")
         return free_space_on_bu_hdd
 
+    def wait_for_ntp_update(self, timeout_seconds: float) -> None:
+        start = time()
+        while not self._system_clock_synchronized_with_ntp():
+            if time() - start > timeout_seconds:
+                raise TimeSynchronisationError(f"Waiting for NTP Update timed out (timeout = {timeout_seconds}s)")
+            sleep(0.5)
+
     @staticmethod
-    def system_clock_synchronized_with_ntp() -> bool:
-        timedate_status_raw = subprocess.check_output("timedatectl")
-        return "System clock synchronized: yes" in timedate_status_raw.decode()
+    def _system_clock_synchronized_with_ntp() -> bool:
+        timedate_status_raw = subprocess.check_output("timedatectl").decode()
+        if "System clock synchronized: " not in timedate_status_raw:
```

Comment:
I'm being picky, but the function would still always return `False` if they ever changed the output to `System clock synchronized: 0/1` or `System clock synchronized: YES/NO` or if they localized (provide different translations for different default languages on target systems) the output...

Suggestion:
```py
if "System clock synchronized: yes" in timedate_status_raw:
    return True
elif "System clock synchronized: no" in timedate_status_raw:
    return False
else:
    raise TimeSynchronisationError("NTP synchronisation status cannot be obtained!")
```

Another suggestion (about six times slower, though):
```py
match = re.search("System clock synchronized: (yes|no)", timedate_status_raw)
try:
    return "yes" in match.string
except AttributeError as e:
    raise TimeSynchronisationError("NTP synchronisation status cannot be obtained!") from e
```

Reply:
Haha yeah, be picky, that's allright :D